From 61dff3402b38f849c1ed5d1d046897f4585600a5 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Thu, 4 Jun 2015 14:18:03 -0400
Subject: [PATCH 974/987] fsl_usdpaa: Ensure map name is initialized and
 cleared

Make sure that map name is null terminated when a memory map
is split and is cleared when a map is destroyed

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
Change-Id: If779b54817b9c2d49d6e18106b333a51ca2b2dcf
Reviewed-on: http://git.am.freescale.net:8181/37436
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Honghua Yin <Hong-Hua.Yin@freescale.com>
[Yang: Original patch taken from
QorIQ-SDK-V1.8-SOURCE-20150619-yocto.iso]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index 1f6df29..f971df2 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -230,6 +230,7 @@ static struct mem_fragment *split_frag(struct mem_fragment *frag)
 	x[0]->refs = x[1]->refs = x[2]->refs = 0;
 	x[0]->root_len = x[1]->root_len = x[2]->root_len = frag->root_len;
 	x[0]->root_pfn = x[1]->root_pfn = x[2]->root_pfn = frag->root_pfn;
+	x[0]->name[0] = x[1]->name[0] = x[2]->name[0] = 0;
 	list_add_tail(&x[0]->list, &frag->list);
 	list_add_tail(&x[1]->list, &x[0]->list);
 	list_add_tail(&x[2]->list, &x[1]->list);
@@ -241,9 +242,10 @@ static __maybe_unused void dump_frags(void)
 	struct mem_fragment *frag;
 	int i = 0;
 	list_for_each_entry(frag, &mem_list, list) {
-		pr_info("FRAG %d: base 0x%llx pfn_base 0x%lx len 0x%llx root_len 0x%llx root_pfn 0x%lx refs %d\n",
+		pr_info("FRAG %d: base 0x%llx pfn_base 0x%lx len 0x%llx root_len 0x%llx root_pfn 0x%lx refs %d name %s\n",
 			i, frag->base, frag->pfn_base,
-			frag->len, frag->root_len, frag->root_pfn, frag->refs);
+			frag->len, frag->root_len, frag->root_pfn,
+			frag->refs, frag->name);
 		++i;
 	}
 }
@@ -1141,6 +1143,7 @@ map_match:
 		current_frag = list_entry(current_frag->list.prev,
 					  struct mem_fragment, list);
 	}
+	map->root_frag->name[0] = 0;
 	list_del(&map->list);
 	compress_frags();
 	spin_unlock(&mem_lock);
-- 
1.9.1

